keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,855 @@
+
+- [x] Metrics
+  - Metric name:
+    - `scheduler_scheduling_attempts_total`
+    - `scheduler_framework_extension_point_duration_seconds`
+    - Components exposing the metric: `kube-scheduler`
+  - Metric name:
+    - `kube_pod_status_phase`
+    - `kube_pod_status_scheduled_time`
+    - Components exposing the metric: `kube-apiserver`
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Yes, a new metrics:
+
+- `scheduler_numeric_taint_evaluations_total`: tracks each numeric evaluation with its result.
+- `scheduler_numeric_tolerations_total`: tracks successful scheduling with numeric tolerations.
+These metrics provide visibility into:
+
+1. How frequently the numeric toleration feature is being used
+2. The effectiveness of numeric taint/toleration matching
+3. Per-profile usage patterns for multi-scheduler setups
+
+In addition, the scheduler has an existing `scheduler_unschedulable_pods` metric that handles the multiple failure reasons by incrementing for each plugin that rejects a pod.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+N/A
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No, the feature is designed to be an enhancement to existing logic without introducing any new API communication patterns.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Potentially yes, but the impact should be **minimal**. The numeric toleration operators feature could slightly increase time for operations covered by existing SLIs/SLOs due to integer parsing overhead and validation overhead.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+Same as existing taint/toleration system which is graceful degradation.
+
+###### What are other known failure modes?
+
+A failure mode due to numeric toleration operators have integer parsing errors from malformed taint/toleration values causing pods to be rejected with clear error messages.
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+- 2025-08-11: Initial KEP
+
+## Drawbacks
+
+## Alternatives
```

Comment:
Another possible alternative here.. Just adding a new field `NumValue int64` to Taint/Toleration, which will solve many discussed concerns such as parsing error, performance degradation, etc. I'm not sure if having `Value string` and `NumValue int64` is a beautiful design though

cc @macsko @dom4ha

Reply:
This approach has some downsides besides th not beautiful design, i.e. users might set wrong field or both fields accidentally, complex validation logic, memory/storage overhead.

Reply:
Added.